### ISSUE TITLE
network inspect doesn't correctly display attachable information in 1.13 rc2

### ISSUE THREAD

**Issue by Michael-Hamburger** [docker/docker#28879]
**Description**

I updated my docker 1.12 to 1.13 rc2 to be able to use the attachable feature for networks (https://github.com/docker/docker/pull/25962). I deleted my old network and recreated it with --attachable option. docker network inspect shows attachable true. After I added a container to the network attachable is set to false.

**Steps to reproduce the issue:**
1. Create the network
``` 
   docker network create \
   --driver overlay \
   --attachable \
    zabbix-network
```
2. Display network inspect
``` 
   docker network inspect zabbix-network
```
3. Verify that Attachable is true (The value for created is wrong?)
``` 
   ...
   "Created": "0001-01-01T00:00:00Z",
   "Attachable": true,
   ...
```
4. Create a container and add it to the network
``` 
    docker run --name postgres --env POSTGRES_PASSWORD=0815rootpw! -d --network zabbix-network zabbix/zabbix-server-pgsql:alpine-3.2-latest
```
5. Display network inspect
``` 
   docker network inspect zabbix-network
```

**Describe the results you received:**
``` 
 "Attachable": false,
        "Containers": {
            "1fc9f2eea5ec3a5d4ba8b913d7e86768ed4200ca770acb01f24654865876c52c": {
                "Name": "postgres",
                ...
            }
        },
```

**Describe the results you expected:**
``` 
 "Attachable": true,
        "Containers": {
            "1fc9f2eea5ec3a5d4ba8b913d7e86768ed4200ca770acb01f24654865876c52c": {
                "Name": "postgres",
                ...
            }
        },
```

**Additional information you deem important (e.g. issue happens only occasionally):**
I am not sure but I think when using docker service create with --network zabbix-network the container isn't added to the network inspect output.  I also have other containers running so don't panic when seeing that three containers are running.

**Output of `docker version`:**

```
Client:
 Version:      1.13.0-rc2
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   1f9b3ef
 Built:        Wed Nov 23 06:17:45 2016
 OS/Arch:      linux/amd64

Server:
 Version:             1.13.0-rc2
 API version:         1.25
 Minimum API version: 1.12
 Go version:          go1.7.3
 Git commit:          1f9b3ef
 Built:               Wed Nov 23 06:17:45 2016
 OS/Arch:             linux/amd64
 Experimental:        false
```

**Output of `docker info`:**

```
Containers: 8
 Running: 3
 Paused: 0
 Stopped: 5
Images: 38
Server Version: 1.13.0-rc2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 106
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: 2roi10et2p908xc60w8ap8i6x
 Is Manager: true
 ClusterID: 69a3wiv7ja5bxs7isbndnmglv
 Managers: 1
 Nodes: 2
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 5.9.24.226
 Manager Addresses:
  5.9.24.226:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 51371867a01c467f08af739783b8beafc154c4d7
init version: 949e6fa
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.62 GiB
Name: bottle.haembi.de
ID: W4ZV:L457:5MAK:WYYG:UE3Y:ECG6:A5UI:7EFA:IW7Q:5BE2:UPBP:G5YW
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by vdemeester** [comment]
@Michael-Hamburger I can confirm and reproduce. If any container is connected to the *attachable* network, it's shown as not attachable anymore… As soon no container are attached, it's back to attachable. It's the same if the container have been created by services.

**Comment by thaJeztah** [comment]
partial duplicate of https://github.com/docker/docker/issues/28817 ?

**Comment by Michael-Hamburger** [comment]
@vdemeester When I create a new network and only add container via service at first the container isn't shown.

``` 
hamburml@bottle ~ $ docker service create --name postgresservice --env POSTGRES_PASSWORD=0815rootpw! --network testnetwork postgres:9.6.1-alpine
eaeixhnz108pdjdmoe1afu1vm
hamburml@bottle ~ $ docker network inspect testnetwork
[
    {
        "Name": "testnetwork",
        "Id": "e10lshtkx4hia8ngicvu8ydav",
        "Created": "0001-01-01T00:00:00Z",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": []
        },
        "Internal": false,
        "Attachable": true,
        "Containers": null,
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "4099"
        },
        "Labels": null
    }
]

```

**Comment by vdemeester** [comment]
@Michael-Hamburger maybe your container was not yet created though because it definitely should show the container (at least on the same node…)

**Comment by Michael-Hamburger** [comment]
@vdemeester Correct. The container is shown on the same node.

**Comment by thaJeztah** [comment]
Actually, could be duplicate of https://github.com/docker/docker/issues/27124 (fixed through https://github.com/docker/docker/pull/28721)

**Comment by sanimej** [comment]
@Michael-Hamburger 
`When I create a new network and only add container via service at first the container isn't shown.` - `network inspect` shows only the containers that are running on that node. So this is expected if the service's task wasn't running on that node.

@thaJeztah The attachable status being incorrect is an UX issue and it will be fixed through #28721 as you noted.  This issue can be closed.

**Comment by thaJeztah** [comment]
Thanks @sanimej let me close this one, then
